tests/gpt_j.rs

```diff
@@ -0,0 +1,236 @@
+use std::path::PathBuf;
+
+use rust_bert::gpt_j::{
+    GptJConfig, GptJConfigResources, GptJLMHeadModel, GptJMergesResources, GptJModelResources,
+    GptJVocabResources,
+};
+use rust_bert::pipelines::common::ModelType;
+use rust_bert::pipelines::generation_utils::{Cache, LMHeadModel};
+use rust_bert::pipelines::text_generation::{TextGenerationConfig, TextGenerationModel};
+use rust_bert::resources::{LocalResource, RemoteResource, ResourceProvider};
+use rust_bert::Config;
+use rust_tokenizers::tokenizer::{Gpt2Tokenizer, MultiThreadedTokenizer, Tokenizer};
+use rust_tokenizers::vocab::Gpt2Vocab;
+use tch::{nn, Device, Tensor};
+
+/// Equivalent Python code:
+///
+/// ```python
+/// import torch
+/// from transformers import AutoTokenizer, GPTJForCausalLM
+///
+/// device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+///
+/// model = GPTJForCausalLM.from_pretrained("EleutherAI/gpt-j-6B", revision="float16").to(device)
+///
+/// tokenizer = AutoTokenizer.from_pretrained("EleutherAI/gpt-j-6B", padding_side="left")
+/// tokenizer.add_special_tokens({"pad_token": tokenizer.eos_token})
+///
+/// prompts = ["It was a very nice and sunny", "It was a gloom winter night, and"]
+/// inputs = tokenizer(prompts, return_tensors="pt", padding=True, truncation=True).to(device)
+///
+/// with torch.no_grad():
+///     gen_tokens = model.generate(
+///         **inputs,
+///         min_length=0,
+///         max_length=32,
+///         do_sample=False,
+///         early_stopping=True,
+///         num_beams=1,
+///         num_return_sequences=1
+///     )
+///
+/// gen_texts = tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)
+/// ````
+///
+/// To run this test you need to download `pytorch_model.bin` from [EleutherAI GPT-J 6B
+/// (float16)][gpt-j-6B-float16] and then convert its weights with:
+///
+/// ```
+/// python utils/convert_model.py resources/gpt-j-6B-float16/pytorch_model.bin
+/// ```
+///
+/// [gpt-j-6B-float16]:https://huggingface.co/EleutherAI/gpt-j-6B/tree/float16
+#[test]
+#[cfg_attr(not(feature = "all-tests"), ignore)]
+fn gpt_j_generation() -> anyhow::Result<()> {
```

Comment:
If these weights are not available could we please remove this from integration tests? This could be added as an example instead if you wish to be run manually.

You could also try loading the `pytorch_model.bin` model file with the ltest `tch-rs` [update](https://github.com/LaurentMazare/tch-rs/commit/75074f3253d9cc4d5f2988fc03ad16b6101b90d5). It does not work for all model files yet, but it compatible with GPT-J files this would allow readily loading the model from pretrained resources.

Reply:
It looks like I also get a "Expected GenericDict but got None" error as mentioned in [tch-rs/issues/595](https://github.com/LaurentMazare/tch-rs/issues/595#issuecomment-1364108766)

Reply:
Ah that's unfortunate, thank you for checking anyway. It may make sense to move this code snippet to the examples if the download/conversion cannot be easily automated - what do you think?

Reply:
@lerouxrgd it looks like the PR is almost ready to be merged -- any chance this could be moved to an example? This would give more visibility to this new feature, and keep all tests independent of pre-download/conversion steps

Reply:
That's exactly what I was doing :)

Reply:
Done (And I have re-checked that it works fine on a T4 GPU)